pkg/controller/installation/core_controller.go

```diff
@@ -1783,3 +1779,35 @@ func removeInstallationFinalizer(i *operator.Installation) {
 		i.SetFinalizers(stringsutil.RemoveStringInSlice(CalicoFinalizer, i.GetFinalizers()))
 	}
 }
+
+func (r *ReconcileInstallation) updateInstallationDegraded(reason string, err error, instance *operator.Installation, ctx context.Context, log logr.Logger) {
+	r.SetDegraded(reason, err, log)
+	conditions := []operator.TigeraStatusCondition{
+		{Type: operator.ComponentDegraded, Status: operator.ConditionTrue, Reason: reason, Message: err.Error()},
+		{Type: operator.ComponentAvailable, Status: operator.ConditionFalse},
+		{Type: operator.ComponentProgressing, Status: operator.ConditionFalse},
+	}
+
+	for _, condition := range conditions {
+		found := false
+		for i, c := range instance.Status.Conditions {
+			if c.Type == condition.Type {
+				// If the status has changed, update the transition time.
+				condition.LastTransitionTime = c.LastTransitionTime
+				if c.Status != condition.Status {
+					condition.LastTransitionTime = metav1.NewTime(time.Now())
+				}
+				instance.Status.Conditions[i] = condition
+				found = true
+			}
+		}
+		if !found {
+			condition.LastTransitionTime = metav1.NewTime(time.Now())
+			instance.Status.Conditions = append(instance.Status.Conditions, condition)
+		}
+	}
+
+	if err := r.client.Status().Update(ctx, instance); err != nil {
+		log.WithValues("reason", err).Info("Failed to update Installation status")
```

Comment:
I think the signature of the method should change, such that you return `reconcile.Result{}, err` .
And then in the reconcile(..) method, you call 
```
 return r.SetDegraded(...) 
```